src/test/java/uk/ac/ebi/eva/pipeline/model/converters/data/VariantToMongoDbObjectConverterTest.java

```diff
@@ -0,0 +1,144 @@
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.convert.support.GenericConversionService;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Test {@link VariantToMongoDbObjectConverter}
+ */
+@RunWith(ParallelRunner.class)
+public class VariantToMongoDbObjectConverterTest {
+
+    private GenericConversionService conversionService = new GenericConversionService();
+    private Converter variantToMongoDbObjectConverter;
+
+    private final String fileId = "fileId";
+    private final String studyId = "studyId";
+    private final Map<String, Integer> samplesPosition = new LinkedHashMap<>();
+    private final VariantStorageManager.IncludeSrc includeSrc = VariantStorageManager.IncludeSrc.FIRST_8_COLUMNS;
+
+    private boolean includeStats = false;
+    private boolean calculateStats = false;
+    private boolean includeSample = true;
+
+    @Test(expected = IllegalArgumentException.class)
+    public void convertNullVariantShouldThrowAnException(){
+        variantToMongoDbObjectConverter = new VariantToMongoDbObjectConverter(includeStats, samplesPosition,
+                calculateStats, includeSample, includeSrc);
+        variantToMongoDbObjectConverter.convert(null);
+    }
+
+    @Test
+    public void allFieldsOfVariantShouldBeConverted(){
+        String chromosome = "12";
+        int start = 3;
+        int end = 4;
+        String reference = "A";
+        String alternate = "T";
+
+        Variant variant = buildVariant(chromosome, start, end, reference, alternate, fileId, studyId);
+
+        variantToMongoDbObjectConverter = new VariantToMongoDbObjectConverter(includeStats, samplesPosition,
+                calculateStats, includeSample, includeSrc);
+        conversionService.addConverter(variantToMongoDbObjectConverter);
+
+        DBObject dbObject = conversionService.convert(variant, DBObject.class);
+
+        assertNotNull(dbObject);
+        BasicDBObject addToSet = (BasicDBObject) dbObject.get("$addToSet");
+        BasicDBObject files = (BasicDBObject) addToSet.get("files");
+        assertTrue(fileId.equals(files.get("fid")));
+        assertTrue(studyId.equals(files.get("sid")));
+        assertNull(addToSet.get("st"));
+        assertNull(addToSet.get("ids"));
+
+        BasicDBObject setOnInsert = (BasicDBObject) dbObject.get("$setOnInsert");
+        assertTrue(String.format("%s_%s_%s_%s", chromosome, start, reference, alternate).equals(setOnInsert.get("_id")));
+        assertTrue(chromosome.equals(setOnInsert.get("chr")));
+        assertEquals(start, setOnInsert.get("start"));
+        assertEquals(end, setOnInsert.get("end"));
+        assertTrue(reference.equals(setOnInsert.get("ref")));
+        assertTrue(alternate.equals(setOnInsert.get("alt")));
+    }
+
+    @Test
+    public void includeStatsTrueShouldIncludeStatistics() {
```

Comment:
Yes, because it's not clear under which conditions is each field written or not. Setting the boolean values inside the test methods instead of having a default value in the attributes would also help.